src/py/flwr/common/message.py

```diff
@@ -283,22 +285,22 @@ def _create_reply_metadata(self, ttl: str) -> Metadata:
     def create_error_reply(
         self,
         error: Error,
-        ttl: str,
+        ttl: float,
     ) -> Message:
         """Construct a reply message indicating an error happened.
 
         Parameters
         ----------
         error : Error
             The error that was encountered.
-        ttl : str
+        ttl : int
             Time-to-live for this message.
         """
         # Create reply with error
         message = Message(metadata=self._create_reply_metadata(ttl), error=error)
         return message
 
-    def create_reply(self, content: RecordSet, ttl: str) -> Message:
+    def create_reply(self, content: RecordSet, ttl: float = DEFAULT_TTL) -> Message:
```

Comment:
Shouldn't the default be the TTL of the instruction message?

We'd need to make `ttl` an optional argument here and also add a check below that sets `ttl` to the TTL of the instruction message if the user does not set the `ttl` argument. We should probably also check that a user-provided TTL does not exceed the instruction message TTL.

```suggestion
    def create_reply(self, content: RecordSet, ttl: Optional[float] = None) -> Message:
```

Reply:
my understanding was that the`SuperNode`-side `TTL` was to specify the max duration the `TaskRes` is allowed to remain at the `SuperLink` before it gets pulled by the `ServerApp` (so they are different TTLs)

Reply:
I've made the change to make it optional. But to keep things consistent, in this PR if the `ttl` is unset, it will set `DEFAULT_TTL`. In a follow up PR we'll replace it to what we agreed upon (the remaining time based on message creation and it's ttl from serverapp)